src/witan/gateway/handler.clj

```diff
@@ -112,6 +112,26 @@
                          :kixi.comms.query/results results}))))
 
 ;;
+;;Add partition-key finder here!
+;;exception if there's isn't one
+
+(defn uuid
+  [& _]
+  (str (java.util.UUID/randomUUID)))
+
+(def command-key->partition-key-fn
+  {:kixi.datastore.filestore/create-upload-link uuid
+   :kixi.datastore.schema/create uuid
+   :kixi.datastore.filestore/create-file-metadata :kixi.datastore.metadatastore/id
+   :kixi.datastore.bundles/create-datapack :kixi.datastore.metadatastore/id
+   :kixi.datastore.metadata/sharing-change :kixi.datastore.metadatastore/id
+   :kixi.datastore.metadata/update :kixi.datastore.metadatastore/id})
+
+(defn partition-key-fn
+  [key]
+  (or (key command-key->partition-key-fn)
```

Comment:
if key fn gets the key of a map entry - not sure what map entry we're getting here?  Isn't it getting a whole map?  the key argument isn't getting passed in ?